Makefile

```diff
@@ -164,30 +164,31 @@ ifneq ($(MONITORING), false)
 endif
 
 
-# Qt5 dirs (we can't indent with tabs here)
+# Qt dirs (we can't indent with tabs here)
 ifneq ($(detected_OS),Windows)
- export QT5_LIBDIR := $(shell qmake -query QT_INSTALL_LIBS 2>/dev/null)
- QT5_QMLDIR := $(shell qmake -query QT_INSTALL_QML 2>/dev/null)
- QT5_INSTALL_PREFIX := $(shell qmake -query QT_INSTALL_PREFIX 2>/dev/null)
- QT5_PKGCONFIG_INSTALL_PREFIX := $(shell pkg-config --variable=prefix Qt5Core 2>/dev/null)
- ifeq ($(QT5_INSTALL_PREFIX),$(QT5_PKGCONFIG_INSTALL_PREFIX))
-  QT5_PCFILEDIR := $(shell pkg-config --variable=pcfiledir Qt5Core 2>/dev/null)
+ export QT_LIBDIR := $(shell qmake -query QT_INSTALL_LIBS 2>/dev/null)
+ QT_QMLDIR := $(shell qmake -query QT_INSTALL_QML 2>/dev/null)
+ QT_INSTALL_PREFIX := $(shell qmake -query QT_INSTALL_PREFIX 2>/dev/null)
+ QT_MAJOR_VERSION := $(shell qmake -query QT_VERSION | head -c 1 2>/dev/null)
+ QT_PKGCONFIG_INSTALL_PREFIX := $(shell pkg-config --variable=prefix Qt"$(QT_MAJOR_VERSION)"Core 2>/dev/null)
+ ifeq ($(QT_INSTALL_PREFIX),$(QT_PKGCONFIG_INSTALL_PREFIX))
+  QT_PCFILEDIR := $(shell pkg-config --variable=pcfiledir Qt6Core 2>/dev/null)
  else
-  QT5_PCFILEDIR := $(QT5_LIBDIR)/pkgconfig
+  QT_PCFILEDIR := $(QT_LIBDIR)/pkgconfig
  endif
- # some manually installed Qt5 instances have wrong paths in their *.pc files, so we pass the right one to the linker here
+ # some manually installed Qt instances have wrong paths in their *.pc files, so we pass the right one to the linker here
  ifeq ($(detected_OS),Darwin)
   NIM_PARAMS += -L:"-framework Foundation -framework AppKit -framework Security -framework IOKit -framework CoreServices -framework LocalAuthentication"
   # Fix for failures due to 'can't allocate code signature data for'
   NIM_PARAMS += --passL:"-headerpad_max_install_names"
-  NIM_PARAMS += --passL:"-F$(QT5_LIBDIR)"
+  NIM_PARAMS += --passL:"-F$(QT_LIBDIR)"
 
  else
-  NIM_PARAMS += --passL:"-L$(QT5_LIBDIR)"
+  NIM_PARAMS += --passL:"-L$(QT_LIBDIR)"
  endif
  DOTHERSIDE_LIBFILE := vendor/DOtherSide/build/lib/libDOtherSideStatic.a
  # order matters here, due to "-Wl,-as-needed"
- NIM_PARAMS += --passL:"$(DOTHERSIDE_LIBFILE)" --passL:"$(shell PKG_CONFIG_PATH="$(QT5_PCFILEDIR)" pkg-config --libs Qt5Core Qt5Qml Qt5Gui Qt5Quick Qt5QuickControls2 Qt5Widgets Qt5Svg Qt5Multimedia Qt5WebView Qt5WebChannel)"
+ NIM_PARAMS += --passL:"$(DOTHERSIDE_LIBFILE)" --passL:"$(shell PKG_CONFIG_PATH="$(QT_PCFILEDIR)" pkg-config --libs Qt"$(QT_MAJOR_VERSION)"Core Qt"$(QT_MAJOR_VERSION)"Qml Qt"$(QT_MAJOR_VERSION)"Gui Qt"$(QT_MAJOR_VERSION)"Quick Qt"$(QT_MAJOR_VERSION)"QuickControls2 Qt"$(QT_MAJOR_VERSION)"Widgets Qt"$(QT_MAJOR_VERSION)"Svg Qt"$(QT_MAJOR_VERSION)"Multimedia Qt"$(QT_MAJOR_VERSION)"WebView Qt"$(QT_MAJOR_VERSION)"WebChannel)"
```

Comment:
Ideally we don't want `Widgets` here

Reply:
But I guess we _do_ want WebEngine here, right @alexjba?

Reply:
I don't think we need it here. The direct dependency is on QtWebView.